c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -271,7 +271,50 @@ impl<'tcx> TypeChecker<'tcx, '_> {
                     Label::default()
                 })
             }
-
+            Rvalue::Cast(CastKind::PointerFromExposedAddress, ref op, _ty) => {
+                // We support only one case here, which is the case of null pointers
+                // constructed via casts such as `0 as *const T`
+                if let Some(0) = op
+                    .constant()
+                    .and_then(|c| c.literal.try_to_scalar_int())
+                    .and_then(|s| s.try_to_machine_usize(self.tcx).ok())
```

Comment:
See #866.  I'm not sure this is totally correct and there seems to be a simpler implementation possible.

Reply:
Utilizing your changes now